pkg/cli/cmds/agent_service_windows.go

```diff
@@ -0,0 +1,132 @@
+// +build windows
+
+package cmds
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	"github.com/rancher/k3s/pkg/version"
+	"github.com/urfave/cli"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+const (
+	defaultServiceDescription = "Rancher Kubernetes Engine v2 (agent) see docs https://github.com/rancher/rke2#readme"
+)
+
+var serviceSubcommand = cli.Command{
+	Name:   "service",
+	Usage:  "Manage RKE2 as a Windows Service",
+	Action: serviceAction,
+	Flags: []cli.Flag{
+		cmds.ConfigFlag,
+		&cli.BoolFlag{
+			Name:  "add",
+			Usage: "add RKE2 as a Windows Service",
+		},
+		&cli.BoolFlag{
+			Name:  "delete",
+			Usage: "stop and delete RKE2 as a Windows Service",
+		},
+		&cli.StringFlag{
+			Name:  "service-name",
+			Usage: "name for the RKE2 service",
+			Value: version.Program,
+		},
+	},
+}
+
+func serviceAction(ctx *cli.Context) error {
+	add := ctx.Bool("add")
+	del := ctx.Bool("delete")
+	if (!add && !del) || (add && del) {
+		return errors.New("service subcommand requires one of --add or --delete")
+	}
+
+	if del {
+		return deleteWindowsService(ctx.String("service-name"))
+	}
+
+	return addWindowService(ctx.String("service-name"), ctx.String("config"))
+}
+
+func getServicePath() (string, error) {
+	p, err := exec.LookPath(os.Args[0])
+	if err != nil {
+		return "", err
+	}
+	return filepath.Abs(p)
+}
+
+// inspiration https://github.com/containerd/containerd/blob/main/cmd/containerd/command/service_windows.go#L127
+func addWindowService(serviceName, config string) error {
+	p, err := getServicePath()
+	if err != nil {
+		return err
+	}
+	m, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+
+	s, err := m.CreateService(serviceName, p, mgr.Config{
+		ServiceType:  windows.SERVICE_WIN32_OWN_PROCESS,
+		StartType:    mgr.StartAutomatic,
+		ErrorControl: mgr.ErrorNormal,
+		DisplayName:  version.Program,
+		Description:  defaultServiceDescription,
+	}, "agent", "--config "+config)
```

Comment:
should be `"--config=config"` or `"--config", "config"` but not `--config config`
```suggestion
	}, "agent", "--config="+config)
```